packages/playwright/src/AxeCoreWrapper/DefaultAxeCoreWrapper.cs

```diff
@@ -203,44 +208,48 @@ bool iframes
                 return partialResults;
             }
 
-            var frameContexts = await GetFrameContexts(frame, context);
+            var frameContexts = await GetFrameContexts(frame, context).ConfigureAwait(false);
+            var partialResultTasks = new List<Task<List<object?>>>();
             foreach (var fContext in frameContexts)
             {
                 try
                 {
                     object frameContext = JsonConvert.SerializeObject(fContext.Context);
                     string frameSelector = JsonConvert.SerializeObject(fContext.Selector);
-                    var frameHandle = await frame.EvaluateHandleAsync(EmbeddedResourceProvider.ReadEmbeddedFile("shadowSelect.js"), frameSelector);
+                    var frameHandle = await frame.EvaluateHandleAsync(await EmbeddedResourceProvider.ReadEmbeddedFileAsync("shadowSelect.js").ConfigureAwait(false), frameSelector).ConfigureAwait(false);
                     if (frameHandle != null)
                     {
                         var childFrameElement = frameHandle.AsElement();
                         if (childFrameElement == null)
                         {
-                            partialResults.Add(null);
+                            partialResults.Add(Task.FromResult<List<object?>>(new List<object?>(null)));
```

Comment:
All of the error handling lines of this form have three issues:

* They should be `Add`ing to `partialResultTasks`, not `partialResults`
* The error marker lists should be `new List<object?> { null }`, not `new List<object?>(null)` (curly braces vs parens) - the former is a List containing one element (`null`), whereas the latter will just throw an `ArgumentNullException`.
* They are not covered by any tests